src/arrayMethodSort.js

```diff
@@ -4,9 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = defaultCompare) {
+    let countChanges = 0;
+
+    do {
+      countChanges = 0;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        if (compareFunction(this[i], this[i + 1]) > 0) {
+          const temp = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = temp;
```

Comment:
as a useful thing to know, think (or google) how we can swap two values without a third variable. another useful thing would be figuring out how it works :)

Reply:
Changed to destructuring assignment)